core/src/lib.rs

```diff
@@ -0,0 +1,369 @@
+            "ConfigurationAndSWHardeningNeeded".into(),
+        ];
+        policy
+    }
+
+    /// Strict attestation policy intended for production use.
+    pub fn strict_tdx() -> Self {
+        Policy::default()
+    }
+}
+
+/// Configure where and how the attestation request is sent after TLS is established.
+#[derive(Debug, Clone)]
+pub struct AttestationEndpoint {
+    pub path: String,
+    pub host: String,
+    pub use_keep_alive: bool,
+}
+
+impl Default for AttestationEndpoint {
+    fn default() -> Self {
+        Self {
+            path: "/tdx_quote".into(),
+            host: "localhost".into(),
+            use_keep_alive: true,
+        }
+    }
+}
+
+/// Result of the attestation and TLS handshake.
+#[derive(Debug, Clone)]
+pub struct AttestationResult {
+    pub trusted: bool,
+    pub tee_type: TeeType,
+    pub measurement: Option<String>,
+    pub tcb_status: String,
+    pub advisory_ids: Vec<String>,
+}
+
+/// Trait alias for async byte streams on native targets.
+pub trait AsyncByteStream: AsyncRead + AsyncWrite + Unpin + Send {}
+impl<T: AsyncRead + AsyncWrite + Unpin + Send> AsyncByteStream for T {}
+
+/// Verifier that accepts any certificate but records the presented leaf.
+#[derive(Debug, Clone, Default)]
+struct PromiscuousVerifier {
+    last_cert: Arc<Mutex<Option<Vec<u8>>>>,
+}
+
+impl PromiscuousVerifier {
+    fn new() -> Self {
+        Self {
+            last_cert: Arc::new(Mutex::new(None)),
+        }
+    }
+
+    fn take_cert(&self) -> Option<Vec<u8>> {
+        self.last_cert.lock().ok()?.take()
+    }
+}
+
+impl ServerCertVerifier for PromiscuousVerifier {
+    fn verify_server_cert(
+        &self,
+        end_entity: &CertificateDer<'_>,
+        _intermediates: &[CertificateDer<'_>],
+        _server_name: &ServerName<'_>,
+        _ocsp_response: &[u8],
+        _now: UnixTime,
+    ) -> Result<ServerCertVerified, RustlsError> {
+        if let Ok(mut guard) = self.last_cert.lock() {
+            *guard = Some(end_entity.to_vec());
+        }
+        Ok(ServerCertVerified::assertion())
+    }
+
+    fn verify_tls12_signature(
+        &self,
+        _message: &[u8],
+        _cert: &CertificateDer<'_>,
+        _dss: &DigitallySignedStruct,
+    ) -> Result<HandshakeSignatureValid, RustlsError> {
+        Ok(HandshakeSignatureValid::assertion())
+    }
+
+    fn verify_tls13_signature(
+        &self,
+        _message: &[u8],
+        _cert: &CertificateDer<'_>,
+        _dss: &DigitallySignedStruct,
+    ) -> Result<HandshakeSignatureValid, RustlsError> {
+        Ok(HandshakeSignatureValid::assertion())
+    }
+
+    fn supported_verify_schemes(&self) -> Vec<SignatureScheme> {
+        vec![
+            SignatureScheme::ECDSA_NISTP256_SHA256,
+            SignatureScheme::ECDSA_NISTP384_SHA384,
+            SignatureScheme::RSA_PSS_SHA256,
+        ]
+    }
+}
+
+fn get_crypto_provider() -> Arc<CryptoProvider> {
+    Arc::new(aws_lc_rs::default_provider())
+}
```

Comment:
Why are we using this provider? I don't understand what aws_lc_rs is. It might have some assumptions that we don't in our use case, no?